### PR TITLE
chore: migrate to memfs

### LINKED ISSUE #10377: chore(deps): memory-fs is deprecated, migrate to memfs
The test scripts consume [memory-fs](https://github.com/webpack/memory-fs) which should be migrated to [memfs](https://github.com/streamich/memfs).

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
dependency migration

closes #10377

**Did you add tests for your changes?**
Nope

**Does this PR introduce a breaking change?**
Nope

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: test/MultiStats.test.js [hunk 1]
```diff
@@ -15,7 +16,8 @@ describe("MultiStats", () => {
 				entry: "./fixtures/b"
 			}
 		]);
-		compiler.outputFileSystem = new MemoryFs();
+		compiler.outputFileSystem = new createFsFromVolume(new Volume());
+		compiler.outputFileSystem.join = path.join.bind(path);
```
Comment (evilebottnawi) [inline test/MultiStats.test.js]: No need that, it is webpack@5, it is workaround for webpack@4

File: test/Compiler.test.js [hunk 2]
```diff
@@ -269,11 +269,11 @@ describe("Compiler", () => {
 			mode: "production",
 			entry: "./missing",
 			output: {
-				path: "/",
+				path: "/directory",
 				filename: "bundle.js"
 			}
 		});
-		compiler.outputFileSystem = new MemoryFs();
+		compiler.outputFileSystem = new createFsFromVolume(new Volume());
```
Comment (sokra) [inline test/Compiler.test.js]: I guess the leading `new` should be omitted. `createFsFromVolume` sounds like a function and not a class.

Comment (jamesgeorge007) [reply test/Compiler.test.js]: Good catch :clap:

File: test/WatchDetection.test.js [hunk 3]
```diff
@@ -72,7 +72,9 @@ describe("WatchDetection", () => {
 						filename: "bundle.js"
 					}
 				});
-				const memfs = (compiler.outputFileSystem = new MemoryFs());
+				const memfs = (compiler.outputFileSystem = createFsFromVolume(
```
Comment (sokra) [inline test/WatchDetection.test.js]: This test case is broken now. Probably for the same reason as the other test case was broken -> `path` can't be root

**Comment by sokra** [comment]
Thanks

**Pull request merged**